nemo/utils/exp_manager.py

```diff
@@ -307,9 +308,21 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
             exp_dir, name, and version.
     """
     # Add rank information to logger
-    # Note: trainer.global_rank and trainer.is_global_zero are not set until trainer.fit, so have to hack around it
     local_rank = int(os.environ.get("LOCAL_RANK", 0))
-    global_rank = trainer.node_rank * trainer.num_devices + local_rank
+
+    # the exp manager calls some operations that require explicit
+    # synchronization, therefore we need to initialize the process
+    # group to initiate a barrier
+    if parallel_state.is_unitialized():
+
+        def dummy():
+            return
+
+        if trainer.strategy.launcher is not None:
+            trainer.strategy.launcher.launch(dummy, trainer=trainer)
+        trainer.strategy.setup_environment()
+
+    global_rank = torch.distributed.get_rank()
```

Comment:
This forces even single gpu runs to launch torch.distributed, let's only do this if there's more than one gpu

Reply:
07ff8ee4e1c6bf3ac75ba68ece1716aeec2ba921 only updated from before if we initialized the process group